core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -385,9 +385,12 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
 // For backward compatibility
 namespace Kokkos {
 namespace Experimental {
-using Kokkos::Iterate;
-using Kokkos::MDRangePolicy;
-using Kokkos::Rank;
+using Iterate KOKKOS_DEPRECATED = Kokkos::Iterate;
+template <typename... Properties>
+using MDRangePolicy KOKKOS_DEPRECATED = Kokkos::MDRangePolicy<Properties...>;
+template <unsigned N, Kokkos::Iterate OuterDir = Kokkos::Iterate::Default,
+          Kokkos::Iterate InnerDir = Kokkos::Iterate::Default>
+using Rank KOKKOS_DEPRECATED = Kokkos::Rank<N, OuterDir, InnerDir>;
```

Comment:
These one I would want to discuss first

Reply:
Can you briefly say why you have concerns about having them outside the `Impl` namespace before I go ahead and split this pull request?

Reply:
This is `Experimental::` not `Impl::`

Reply:
Fair enough. Now, hasn't `MDRangePolicy` been used a lot to not be considered `Experimental` anymore? Or rather, why is it problematic to deprecate the aliases in the `Experimental` namespace if they have been quite some time in the regular `Kokkos` namespace already?

Reply:
I think its fine, lets mark the Experimental as deprecated.